modules/ngscheckmate/ncm/main.nf

```diff
@@ -0,0 +1,37 @@
+process NGSCHECKMATE_NCM {
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::ngscheckmate=1.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ngscheckmate:1.0.0--py27r41hdfd78af_0':
+        'quay.io/biocontainers/ngscheckmate:1.0.0--py27r41hdfd78af_0' }"
+
+    input:
+    path files
+    path snp_bed
+    val bam_mode
+
+    output:
+    path "*.pdf"                  , emit: pdf
+    path "*.txt"                  , emit: txt
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "output"
+    def mode_flag = bam_mode ? "-B" : "-V"
```

Comment:
Add to `args` instead if it's not provided via `args`

Reply:
Doesn't it have to be provided when given as a separate channel?

Reply:
Yes, but you should check `args` so it's not accidentally provided from there too.:
e.g. the workflow sets the mode_flag to be true, but args were copy-pasted to the `modules.config` accidentally containing the `-V` option. Hopefully the program doesn't fall over then when both `-B -B` or `-B -V` are part of the commandline options.

Reply:
Ok, how should I do that? `-B -V` gives an error, they are mutually exclusive. `-B -B` or `-V -V` seems fine.

Reply:
Can I require it to be passed via `args`, instead of the static channel?

Reply:
I would add something like this in the code:
```nextflow
script:
opts = args.tokenize()
if (files.every{ it.toString().endsWith('.bam') } ) {
    if (!opts.contains('-B')) args += ' -B'
    assert !opts.contains('-V')
} else if ( files.every{ it.toString().endsWith('.vcf.gz') }) {
    if (!opts.contains('-V')) args += ' -V'
    assert !opts.contains('-B')
} else {
    throw new Exception("Files must be of the same type and compressed")
}
"""
for ...
done
ncm.py ...
"""
```
and not have the flag.

Reply:
Thanks. I've added code based on this, including having a variable for whether to unzip vcf files.
I'm just waiting for the updated singularity image, then should be able to add a test files for the bam test and good to go.